mathics/core/element.py

```diff
@@ -337,10 +337,16 @@ def get_rules_list(self):
             rules.append(rule)
         return rules
 
-    def get_sequence(self):
-        """Convert's a Mathics Sequence into a Python's list of elements"""
+    def get_sequence(self) -> Union[tuple, list]:
+        """If self has return that, otherwise turn self into a
+        tuple() and return that"""
+
         from mathics.core.symbols import SymbolSequence
 
+        # FIXME: using the below test causes:
+        # TypeError: boxes_to_text() takes 1 positional argument but 2 were given
```

Comment:
which test?

Reply:
> which test?

```
pyston -m pytest test -x
=========================================== test session starts ============================================
platform linux -- Python 3.8.12, pytest-7.1.2, pluggy-1.0.0
rootdir: /src/external-vcs/github/Mathics3/mathics-core
collected 0 items / 1 error                                                                                

================================================== ERRORS ==================================================
__________________________________ ERROR collecting test/test_calculus.py __________________________________
test/test_calculus.py:6: in <module>
    from .helper import check_evaluation, session
test/helper.py:6: in <module>
    session = MathicsSession()
mathics/session.py:58: in __init__
    self.reset(add_builtin, catch_interrupt)
mathics/session.py:64: in reset
    self.definitions = Definitions(add_builtin)
mathics/core/definitions.py:137: in __init__
    autoload_files(self, ROOT_DIR, "autoload")
mathics/core/definitions.py:56: in autoload_files
    Expression(SymbolGet, String(path)).evaluate(Evaluation(defs))
mathics/core/expression.py:491: in evaluate
    expr, reevaluate = expr.rewrite_apply_eval_step(evaluation)
mathics/core/expression.py:1242: in rewrite_apply_eval_step
    result = rule.apply(new, evaluation, fully=False)
mathics/core/rules.py:90: in apply
    self.pattern.match(yield_match, expression, {}, evaluation, fully=fully)
mathics/core/pattern.py:351: in match
    self.head.match(yield_head, expression.get_head(), vars, evaluation)
mathics/core/pattern.py:193: in match_symbol
    yield_func(vars, None)
mathics/core/pattern.py:341: in yield_head
    self.get_pre_choices(
mathics/core/pattern.py:489: in get_pre_choices
    yield_func(vars)
mathics/core/pattern.py:318: in yield_choice
    self.match_element(
mathics/core/pattern.py:702: in match_element
    self.get_wrappings(
mathics/core/pattern.py:518: in get_wrappings
    yield_func(items[0])
mathics/core/pattern.py:690: in yield_wrapping
    element.match(
mathics/builtin/patterns.py:1023: in match
    self.pattern.match(yield_func, expression, new_vars, evaluation)
mathics/builtin/patterns.py:1228: in match
    yield_func(vars, None)
mathics/core/pattern.py:687: in match_yield
    yield_func(new_vars, items_rest)
mathics/core/rules.py:62: in yield_match
    new_expression = self.do_replace(expression, vars, options, evaluation)
mathics/core/rules.py:223: in do_replace
    return self.function(evaluation=evaluation, **vars_noctx)
mathics/builtin/files_io/files.py:453: in apply_default
    evaluation.message("General", "stream", filename)
mathics/core/evaluation.py:535: in message
    text = self.format_output(
mathics/core/evaluation.py:476: in format_output
    boxes = result.boxes_to_text(evaluation=self)
mathics/core/element.py:493: in boxes_to_text
    return self.boxes_to_format("text", **options)
mathics/core/element.py:481: in boxes_to_format
    return boxes_to_format(self, format, **options)
mathics/core/formatter.py:116: in boxes_to_format
    return lookup_method(boxes, format)(boxes, **options)
mathics/core/formatter.py:135: in ret_fn
    return boxes_to_method(elements, **opts)
E   TypeError: boxes_to_text() takes 1 positional argument but 2 were given
============================================= warnings summary =============================================
```

Reply:
The error actually happens because another error was raised before formatters are loaded. Check it by applying the patch

```
diff --git a/mathics/core/definitions.py b/mathics/core/definitions.py
index 47d0d0e9..87970609 100644
--- a/mathics/core/definitions.py
+++ b/mathics/core/definitions.py
@@ -108,6 +108,11 @@ class Definitions:
         )
         self.trace_evaluation = False
         self.timing_trace_evaluation = False
+
+        # Inport formats before builtins, in case
+        # we need to show a message.
+        import mathics.format
+
         if add_builtin:
             from mathics.builtin import modules, contribute
             from mathics.settings import ROOT_DIR
@@ -153,7 +158,6 @@ class Definitions:
             self.clear_cache()
 
         # This loads all the formatting functions
-        import mathics.format
 
         # FIXME load dynamically as we do other things
         # import mathics.format.asy  # noqa
```

Reply:
Thanks for tracking this down. My understanding is that `mathics.format` was one of the main things that was getting worked on right now. 

(Not scanner stuff, Precision, or Box formatting of which two of these areas stray a bit outside of the kinds of things you are good at.) 

I would be grateful if you would do the needful here. Thanks.

Reply:
> Thanks for tracking this down. My understanding is that `mathics.format` was one of the main things that was getting worked on right now.
> 
> (Not scanner stuff, Precision, or Box formatting of which two of these areas stray a bit outside of the kinds of things you are good at.)
> 

Again, I am aware of my limitations. However, in order to make work some simple and appealing tests, I need also that these parts work properly. 

> I would be grateful if you would do the needful here. Thanks.

I am on the way. I also put some other PRs in that direction.